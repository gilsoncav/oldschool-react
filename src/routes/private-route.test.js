import { MemoryRouter } from 'react-router-dom';
import PrivateRoute from './private-route';

jest.mock('../service/auth-service.js', () => ({
  ...jest.requireActual('../service/auth-service.js'),
  isUserLogged: jest
    .fn()
    .mockReturnValueOnce(true)
    .mockReturnValueOnce(false),
}));

describe('PrivateRoute component', () => {
  it('renders correctly if user is logged', () => {
    const snapshot = snapRender(
      <MemoryRouter>
        <PrivateRoute component={() => <div>Teste</div>} />
      </MemoryRouter>,
    );
    expect(snapshot).toMatchSnapshot();
  });

  it('renders correctly if user is not logged', () => {
    const snapshot = snapRender(
      <MemoryRouter>
        <PrivateRoute component={() => <div>Teste</div>} />
      </MemoryRouter>,
    );
    expect(snapshot).toMatchSnapshot();
  });
});
