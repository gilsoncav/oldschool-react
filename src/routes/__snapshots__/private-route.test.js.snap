// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateRoute component renders correctly if user is logged 1`] = `
<div>
  Teste
</div>
`;

exports[`PrivateRoute component renders correctly if user is not logged 1`] = `null`;
